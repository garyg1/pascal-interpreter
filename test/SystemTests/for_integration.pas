program for_integration;

procedure a(x, y : integer);
begin
    writeln('procedure. x: ', x, ', y: ', y);
    x := x + 1;
    y := y + 1;
end;

var x, y, z : integer;
begin
    for x := 1 to 3 do
    begin
        for y := 4 to 6 do
        begin
            a(x, y);
            writeln('loop-inner. x: ', x, ', y: ', y);
            
            // z should not be const though
            z := y;
            while z > 4 do
            begin
                z := z - 1;
                writeln('z: ', z);
                continue;
                writeln('should not be reached');
            end;
            if x = 2 then break;
        end;
        writeln('loop-outer. x: ', x, ', y: ', y);
    end;

    
    writeln('x: ', x, ', y: ', y);
end.