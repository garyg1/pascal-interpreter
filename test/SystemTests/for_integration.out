procedure. x: 1, y: 4
loop-inner. x: 1, y: 4
procedure. x: 1, y: 5
loop-inner. x: 1, y: 5
z: 4
procedure. x: 1, y: 6
loop-inner. x: 1, y: 6
z: 5
z: 4
loop-outer. x: 1, y: 6
procedure. x: 2, y: 4
loop-inner. x: 2, y: 4
loop-outer. x: 2, y: 4
procedure. x: 3, y: 4
loop-inner. x: 3, y: 4
procedure. x: 3, y: 5
loop-inner. x: 3, y: 5
z: 4
procedure. x: 3, y: 6
loop-inner. x: 3, y: 6
z: 5
z: 4
loop-outer. x: 3, y: 6
x: 3, y: 6
